### ISSUE TITLE
[Bug] Product.fromJson constructors still using graph structure for sub constructors

### ISSUE THREAD

**Issue by reesz** [imsujan276/shopify_flutter#24]
Appreciate the quick response on the other issue 🙏 
This one is not as breaking, but I felt like reporting it anyway.

A `Product` has two ways of being constructed: `fromJson` and `fromGraphJson`.
The difference between both is that `fromGraphJson` is expecting the json to look like this:
### Graph Format
```json
{
  "node": {
    "title": "Product title",
    ... fields
  }
}
```
and is accessing it's top level fields with `json['node']['title']`.

`fromJson` on the other hand emits the `node` wrapper & expects the json to look like this:
### JSON format
```json
{
  "title": "Product title",
  ... fields
}
```
therefore accessing it's top level fields with `json['title']`.

But when sub-constructors are being called for things like `images`, `option`, `collectionList`, `productVariations` etc.
They expect the json to be in graph format again, `product.dart:135`:
```dart
  static List<ProductVariant> _getProductVariants(Map<String, dynamic> json) {
    ///      \/ --- accessing through the `node` wrapper of a graph response
    if (json['node']?['variants'] == null) return [];
    return ((json['node']?['variants']?['edges'] ?? []) as List)
        .map((v) => ProductVariant.fromGraphJson(v ?? const {}))
        .toList();
  }
```

I guess a simple fix could be something like
```dart
    json = json.containsKey('node') ? json : { 'node': json };
```
at the top of all methods like `_getProductVariants` to wrap the json in a `node` wrapper if not present, so there's no need for two individual methods.

**Comment by imsujan276** [comment]
Thank you for pointing it out. 

But doing 
``` dart
json = json.containsKey('node') ? json : { 'node': json };
```

won't be enough as in some cases, nested edges are also present. In this case, you can see that we are mapping the `edges` of the `node` `variants`. 

Also, other functions are being used for both data types like `_getOptionList`, `_getCollectionList`, and more.

**Comment by imsujan276** [comment]
This issue has been fixed and updated the package to [v1.4.1](https://pub.dev/packages/shopify_flutter)
